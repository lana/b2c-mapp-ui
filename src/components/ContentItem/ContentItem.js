import { ChevronRightIcon, SuccessMicroillustration as Success } from '@lana/b2c-mapp-ui-assets';

import Heading from '../Heading/Heading.vue';
import TextParagraph from '../TextParagraph/TextParagraph.vue';

const components = {
  ChevronRightIcon,
  Success,
  Heading,
  TextParagraph,
};

const props = {
  dataTestId: {
    type: String,
    default: 'content-item',
  },
  title: {
    type: String,
    default: '',
    required: true,
  },
  metaText: {
    type: String,
    default: '',
  },
  hasForwardButton: {
    type: Boolean,
    default: true,
  },
  success: {
    type: Boolean,
    default: false,
  },
  disabled: Boolean,
  noBorder: Boolean,
};

const computed = {
  hasIcon() {
    const result = (this.success || this.hasForwardButton);
    return result;
  },
  icon() {
    const result = (this.hasForwardButton && !this.success) ? 'ChevronRightIcon' : 'Success';
    return result;
  },
  dataTestIdValue() {
    const result = `${(this.success) ? 'success-' : ''}${this.dataTestId}-forward-icon`;
    return result;
  },
};

const methods = {
  emitClickEvent(event) {
    if (this.disabled || this.success) { return; }
    this.$emit('click', event);
  },
};

const ContentItem = {
  computed,
  components,
  props,
  methods,
};

export default ContentItem;
