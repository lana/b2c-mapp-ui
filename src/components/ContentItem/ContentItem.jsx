import PropTypes from 'prop-types';
import { ForwardIcon } from '@lana/b2c-mapp-ui-assets';

import Heading from '../Heading/Heading';
import TextParagraph from '../TextParagraph/TextParagraph';
import CSS from './styles.css';

const ContentItem = ({ mediaColor, media, meta, onClick, className = '', title, dataTestId }) => {
  const mediaColorClass = (mediaColor) ? CSS[mediaColor] : '';
  const mediaIcon = (media) ? (<div data-testid={`${dataTestId}-media-icon`} className={`${CSS.media} ${mediaColorClass}`}>{media}</div>) : '';
  const metaInfo = (meta) ? <TextParagraph dataTestId={`${dataTestId}-meta-text`}>{meta}</TextParagraph> : '';
  const forwardIcon = (onClick) ? <ForwardIcon data-testid={`${dataTestId}-forward-icon`} color="inactive" className={CSS.icon} /> : null;
  const invokeOnClickIfPresent = (event) => {
    if (!onClick) { return; }
    onClick(event);
  };
  return (
    <li data-testid={dataTestId} onClick={invokeOnClickIfPresent} className={`${CSS.item} ${className}`}>
      {mediaIcon}
      <div data-testid={`${dataTestId}-heading`} className={CSS.body}>
        <Heading type="txt-medium">{title}</Heading>
        {metaInfo}
      </div>
      {forwardIcon}
    </li>
  );
};

ContentItem.defaultProps = {
  mediaColor: false,
  media: null,
  meta: null,
  onClick: () => {},
  className: '',
  title: '',
  dataTestId: 'content-item',
};

ContentItem.propTypes = {
  mediaColor: PropTypes.bool,
  media: PropTypes.node,
  meta: PropTypes.node,
  onClick: PropTypes.func,
  className: PropTypes.string,
  title: PropTypes.string,
  dataTestId: PropTypes.string,
};

export default ContentItem;
