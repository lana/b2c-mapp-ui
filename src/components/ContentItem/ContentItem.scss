@import '../../styles/colors';
@import '../../styles/typography';

$item-box-size: 48px;

.item {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
  position: relative;
  height: $item-box-size;
  padding: 11px 16px;

  &.success {
    background-color: $black-200;
    border-radius: 16px;
  }

  &:active,
  &.pressed {
    background-color: $black-200;
  }
}

.title {
  font: $default-m;
  margin-bottom: 2px;
}

.media {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  overflow: hidden;
  margin-right: 16px;
  width: $item-box-size;
  height: 100%;
  border-radius: 14px;
  border: solid 1px $black-300;
}

.no-border {
  border: none;
}

.body {
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  flex-grow: 1;
}

.item-icon {
  flex-shrink: 0;
}

.meta-text {
  font: $tablet-s;
}
