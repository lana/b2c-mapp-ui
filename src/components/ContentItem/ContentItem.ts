import { ChevronRightIcon, SuccessMicroillustration as Success } from '@lana/b2c-mapp-ui-assets';
import { Comment, defineComponent } from 'vue';

import Heading from '../Heading/Heading.vue';
import TextParagraph from '../TextParagraph/TextParagraph.vue';

const ContentItem = defineComponent({
  name: 'ContentItem',
  components: {
    ChevronRightIcon,
    Success,
    Heading,
    TextParagraph,
  },
  props: {
    dataTestId: {
      type: String,
      default: 'content-item',
    },
    title: {
      type: String,
      default: '',
    },
    metaText: {
      type: String,
      default: '',
    },
    hasForwardButton: {
      type: Boolean,
      default: true,
    },
    success: {
      type: Boolean,
      default: false,
    },
    disabled: Boolean,
    noBorder: Boolean,
  },
  emits: ['click'],
  data() {
    return {
      isPressed: false,
    };
  },
  computed: {
    hasDefaultSlot() {
      const result = this.$slots.default && this.$slots.default().findIndex((node) => (node.type !== Comment)) !== -1;
      return result;
    },
    hasIcon() {
      const result = (this.success || this.hasForwardButton);
      return result;
    },
    hasMetaText() {
      const result = (this.metaText || (this.$slots.customMetaText && this.$slots.customMetaText().findIndex((node) => (node.type !== Comment)) !== -1));
      return result;
    },
    iconName() {
      const result = (this.hasForwardButton && !this.success) ? 'ChevronRightIcon' : 'Success';
      return result;
    },
    iconDataTestId() {
      const result = `${(this.success) ? 'success-' : ''}${this.dataTestId}-forward-icon`;
      return result;
    },
  },
  methods: {
    toggleIsPressed() {
      this.isPressed = !this.isPressed;
    },
    emitClickEvent(event: MouseEvent) {
      if (this.disabled || this.success) { return; }
      this.isPressed = false;
      this.$emit('click', event);
    },
  },
});

export default ContentItem;
