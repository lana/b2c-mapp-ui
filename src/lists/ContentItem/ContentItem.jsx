import CSS from './ContentItem.css'
import Text from '../../typography/Text'
import { ForwardIcon } from '@lana/b2c-mapp-ui-assets'

export default ({ mediaColor, media, meta, onClick, className, title }) => {
	const mediaColorClass = mediaColor ? CSS[mediaColor] : ''
	const mediaIcon = media ? <div className={`${CSS.media} ${mediaColorClass}`}>{media}</div> : ''
	const metaInfo = meta ? (
		<Text type={`callout`} color={`concrete`}>
			{meta}
		</Text>
	) : (
		''
	)
	return (
		<li onClick={onClick ? e => onClick(e) : false} className={`${CSS.item} ${className || ''}`}>
			{mediaIcon}
			<div className={CSS.body}>
				<Text>{title}</Text>
				{metaInfo}
			</div>
			{onClick ? <ForwardIcon color="inactive" className={CSS.icon} /> : ''}
		</li>
	)
}
