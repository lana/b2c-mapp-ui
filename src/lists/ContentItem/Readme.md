# ContentItem
A list item which usually takes the consumer to a content in another screen.

| Prop |  Value Type | Required |
| --- | --- | --- |
| `className` | `String` | false | 
| `media` | `ReactNode` | false | 
| `mediaColor` | `blue`, `green`, `yellow`, `brown`, `purple` | false | 
| `title` | `String` | **true** | 
| `meta` | `String` | false | 
| `onClick` | `(e: HTMLMouseEvent) => void` | false | 

```
<ul>
    <ContentItem title='1st item' meta='additional metadata' media={<img src="https://source.unsplash.com/random/44x44" />} />
    <ContentItem title='2nd item' meta='additional metadata' media={<img src="https://source.unsplash.com/random/44x44" />} />
    <ContentItem title='3rd item' meta='additional metadata' media={<img src="https://source.unsplash.com/random/44x44" />} />
</ul>
```




