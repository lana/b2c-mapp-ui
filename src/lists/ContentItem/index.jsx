import CSS from './styles.css'
import Heading from '../../typography/Heading'
import Text from '../../typography/Text'
import { ForwardIcon } from '@lana/b2c-mapp-ui-assets'

export default ({ mediaColor, media, meta, onClick, className, title }) => {
	const mediaColorClass = mediaColor ? CSS[mediaColor] : ''
	const mediaIcon = media ? <div className={`${CSS.media} ${mediaColorClass}`}>{media}</div> : ''
	const metaInfo = meta ? (
		<Text>{meta}</Text>
	) : (
		''
	)
	return (
		<li onClick={onClick ? e => onClick(e) : false} className={`${CSS.item} ${className || ''}`}>
			{mediaIcon}
			<div className={CSS.body}>
				<Heading type='txt-medium'>{title}</Heading>
				{metaInfo}
			</div>
			{onClick ? <ForwardIcon color="inactive" className={CSS.icon} /> : ''}
		</li>
	)
}
