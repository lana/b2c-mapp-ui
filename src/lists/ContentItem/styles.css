.item {
	display: flex;
	flex-direction: row;
	justify-content: flex-start;
	align-items: center;
	overflow: hidden;
	position: relative;
	height: 44px;
	padding: 8px 16px;
}

.media {
	display: flex;
	justify-content: center;
	align-items: center;
	flex-shrink: 0;
	overflow: hidden;
	margin-right: 12px;
	width: 44px;
	height: 100%;
	border-radius: 3px;
	background: var(--black-400);
}

.media.blue-300 { background-color: var(--blue-300); }
.media.yellow-300 { background-color: var(--yellow-300); }
.media.green-300 { background-color: var(--green-300); }
.media.brown-300 { background-color: var(--brown-300); }
.media.purple-300 { background-color: var(--purple-300); }
.media.black-100 { background-color: var(--black-100); }
.media.black-200 { background-color: var(--black-200); }
.media.black-300 { background-color: var(--black-300); }

.body {
	height: 100%;
	display: flex;
	flex-direction: column;
	align-items: start;
	justify-content: center;
	flex-grow: 1;
}

.icon {
	flex-shrink: 0;
}
