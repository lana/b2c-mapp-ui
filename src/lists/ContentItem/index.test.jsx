
import ContentItem from './index'
import { mount, shallow } from 'enzyme';

describe('UI/lists/ContentItem', () => {
	let wrapperS;
	let wrapperM;
	const props = {
		mediaColor: "MEDIACOLOR",
		media: <img src="any"/>,
		meta: "META",
		textColor: "TEXTCOLOR",
		onClick: () => { console.log('test'); return 'bu'},
		className: "CLASSNAME",
		title: "TITLE"
	}
	beforeEach(()=>{
		wrapperS = shallow(<ContentItem {...props} />)
		wrapperM = mount(<ContentItem {...props} />)
	})
	test('should render correctly',  () => {
		expect(wrapperS).toMatchSnapshot()
	})
	test('should render correctly with minimal config',  () => {
		wrapperS = shallow(<ContentItem title="title" />)
		expect(wrapperS).toMatchSnapshot()
	})

	test('should render correctly with params: mediaColor, media, meta, className, title ',  () => {
		const div = document.createElement('div');
		// workaround to avoid preact/enzyme conflict
		div.innerHTML = wrapperM.html()
		expect(div.querySelectorAll(`div.${props.mediaColor}`).length).toBe(1)
		expect(div.querySelector('img').src).toBe('http://localhost/any')
		expect(div.querySelectorAll(`li.${props.className}`).length).toBe(1)
		// expect(div.querySelector(`p.callout`).innerHTML).toContain(props.meta)
		expect(div.querySelectorAll(`li.${props.className}`).length).toBe(1)
	})
	test('should test click behavior',  () => {
		let check = false
		wrapperS = shallow(<ContentItem title="title" onClick={() => { check=true }} />)
		wrapperS.find('[onClick]').simulate('click')
		expect(check).toBe(true)
	})
})
