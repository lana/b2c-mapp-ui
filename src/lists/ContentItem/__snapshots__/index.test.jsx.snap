// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/lists/ContentItem should render correctly 1`] = `ShallowWrapper {}`;

exports[`UI/lists/ContentItem should render correctly with minimal config 1`] = `ShallowWrapper {}`;
